src/components/macro/macro-item/macro-item.component.ts

```diff
@@ -0,0 +1,208 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewChild } from '@angular/core';
+
+import {MacroAction, macroActionType} from '../../../config-serializer/config-items/macro-action/MacroAction';
+
+import {DelayMacroAction} from '../../../config-serializer/config-items/macro-action/DelayMacroAction';
+import {TextMacroAction} from '../../../config-serializer/config-items/macro-action/TextMacroAction';
+
+import {MouseButtonMacroAction} from '../../../config-serializer/config-items/macro-action/MouseButtonMacroAction';
+import {MoveMouseMacroAction} from '../../../config-serializer/config-items/macro-action/MoveMouseMacroAction';
+import {ScrollMouseMacroAction} from '../../../config-serializer/config-items/macro-action/ScrollMouseMacroAction';
+
+import {KeyMacroAction} from '../../../config-serializer/config-items/macro-action/KeyMacroAction';
+
+import {KeyModifiers}  from '../../../config-serializer/config-items/KeyModifiers';
+
+import { MacroActionEditorComponent } from '../macro-action-editor/macro-action-editor.component';
+import { find as _find } from 'lodash';
+
+// Flatten scancodes for easier label retrieval
+// @todo Should this be its own importable file?
+const scancodesJSON = require('json!../../popover/tab/keypress/scancodes.json');
+
+function flattenScancodes(data: any[]) {
+    let output: any[] = [];
+    for (let i = 0, len = data.length; i < len; i++) {
+        const group = data[i];
+        if (group.children.length) {
+            output = output.concat(group.children);
+        }
+    }
+    return output;
+}
+
+const scancodes = flattenScancodes(scancodesJSON);
+
+@Component({
+    selector: 'macro-item',
+    template: require('./macro-item.component.html'),
+    styles: [require('./macro-item.component.scss')],
+    host: { 'class': 'macro-item' }
+})
+export class MacroItemComponent implements OnInit, OnChanges {
+
+    @Input() macroAction: MacroAction;
+    @Input() editable: boolean;
+    @Input() deletable: boolean;
+    @Input() moveable: boolean;
+
+    @Output() save = new EventEmitter<any>();
```

Comment:
They don't seem to emit any value. If that's the case then use `EventEmitter<void>` type.